src/Skia/Perspex.Skia/FormattedTextImpl.cs

```diff
@@ -75,36 +108,206 @@ public Size Measure()
+                    bytes = Encoding.UTF8.GetBytes(subString);
+                    pinnedArray = GCHandle.Alloc(bytes, GCHandleType.Pinned);
+                    pointer = pinnedArray.AddrOfPinnedObject();
+                    // for some reason I have to pass nBytes * 2
+                    measured = (int)Paint.BreakText(pointer, (IntPtr)(bytes.Length * 2), constraint, out lineWidth) / 2;
+
+                    // some weird unicode byte issue again
+                    //if(subString.Length % 2 == 1)
+                    //{
+                    //	measured -= 1;
+                    //}
+
+                    pinnedArray.Free();
+
+                    if (measured == 0)
+                    {
+                        measured = 1;
+                        lineWidth = -1;
+                    }
+
+                    char nextChar = ' ';
+                    if (curOff + measured < length)
+                        nextChar = _text[curOff + measured];
+
+                    if (nextChar != ' ')
+                    {
+                        //Perform scan for the last space and end the line there
+                        for (int si = curOff + measured - 1; si > curOff; si--)
+                        {
+                            if (_text[si] == ' ')
+                            {
+                                measured = si - curOff;
+                                extraSkip = 1;
+                                //Rects[si] = SkRect();
+                                break;
+                            }
+                        }
+                    }
+
+
+                }
+
+                PerspexFormattedTextLine line = new PerspexFormattedTextLine();
+                line.Start = curOff;
+                line.Length = measured;
+                line.Width = lineWidth;
+                line.Height = lineHeight;
+                line.Top = curY;
+
+                if (line.Width < 0)
+                    line.Width = _skiaRects[line.Start + line.Length - 1].Right;
+
+                //Build character rects
+                for (int i = line.Start; i < line.Start + line.Length; i++)
+                {
+                    float prevRight = 0;
+                    if (i != line.Start)
+                        prevRight = _skiaRects[i - 1].Right;
+
+                    subString = _text.Substring(line.Start, i - line.Start + 1);
+                    float w = Paint.MeasureText(subString);
+
+                    SKRect rc;
+                    rc.Left = prevRight;
+                    rc.Right = w;
+                    rc.Top = line.Top;
+                    rc.Bottom = line.Top + line.Height;
+                    _skiaRects[i] = rc;
+                }
+
+                subString = _text.Substring(line.Start, line.Length);
+                line.Width = Paint.MeasureText(subString);
+
+                _skiaLines.Add(line);
+
+                curY += lineHeight; // + mLeading;
```

Comment:
Will clean this up. I was experimenting with getting the correct vertical line separation working.